mlrun/runtimes/base.py

```diff
@@ -763,6 +770,16 @@ def _submit_job(self, runspec, schedule, db, watch):
 
         return self._wrap_run_result(resp, runspec, schedule=schedule)
 
+    @staticmethod
+    def _handle_submit_job_http_error(error: requests.HTTPError):
+        # if we receive a 400 status code, this means the request was invalid and the run wasn't created in the DB.
+        # so we don't need to update the run state and we can just raise the error.
+        # more status code handling can be added here if needed
+        if error.response.status_code == http.HTTPStatus.BAD_REQUEST.value:
+            raise mlrun.errors.MLRunBadRequestError(
+                f"Bad request to mlrun api: {error.response.text}"
+            )
+
```

Comment:
Seems like `BAD_REQUEST` will be suitable to all other client requests to the MLRun API, maybe we should add that exception handling to a low level layer such as the `httpdb.api_call`?